GrandCentralBoard/Widgets/WebsiteAnalytics/WebsiteAnalyticsWidget.swift

```diff
@@ -8,45 +8,111 @@
 import GCBCore
 
 
+private let widgetTitle = "Blogposts".localized.uppercaseString
+
+
 final class WebsiteAnalyticsWidget: WidgetControlling {
+    private enum State {
+        case Loading
+        case Data(reports: [PageViewsRowReport])
+        case NoData
+        case Error
+    }
 
     private let widgetView: TableWidgetView
-    private let widgetViewWrapper: UIView
-    let sources: [UpdatingSource]
+    private let mainView: UIView
 
-    var view: UIView {
-        return widgetViewWrapper
+    private var state: State = .Loading {
+        didSet {
+            changeFromState(oldValue, toState: state)
+        }
     }
 
     init(sources: [UpdatingSource]) {
         self.widgetView = TableWidgetView.fromNib()
         self.sources = sources
 
-        let viewModel = WidgetTemplateViewModel(title: "BLOGPOSTS", subtitle: "MOST POPULAR", contentView: widgetView)
+        let viewModel = WidgetTemplateViewModel(title: widgetTitle,
+                                                subtitle: "Most popular".localized.uppercaseString,
+                                                contentView: widgetView)
         let layoutSettings = WidgetTemplateLayoutSettings(contentMargin: UIEdgeInsets(top: -90, left: 0, bottom: 0, right: 0))
-        widgetViewWrapper = WidgetTemplateView.viewWithViewModel(viewModel, layoutSettings: layoutSettings)
+        let widgetViewWithHeader = WidgetTemplateView.viewWithViewModel(viewModel, layoutSettings: layoutSettings)
```

Comment:
I did the initial implementation on purpose - to expose and highlight the relationship between views. I'm not sure if widget should contain its error in its own hierarchy. Because I have no good arguments for it, I did the suggested change to unify my code with Bonus widget's implementation.